src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -590,18 +1012,99 @@ def _create_default_1d_data(self):
+        the case of .pdb files being used this version will contain more complete data.
+
+        :return: The full sld data created from the various different sources
+        :rtype: MagSLD
+        """
+        #CARRY OUT COMPATIBILITY CHECK - ELSE RETURN None
+        # Set default data when nothing loaded yet
+        omfdata = sas_gen.OMFData()
+        #load in user chosen position data
+        #If no file given this will be used to generate the position data
+        if (not self.is_mag) and (not self.is_nuc):
+            omfdata.xnodes = int(self.txtXnodes.text())
+            omfdata.ynodes = int(self.txtYnodes.text())
+            omfdata.znodes = int(self.txtZnodes.text())
+            omfdata.xstepsize = float(self.txtXstepsize.text())
+            omfdata.ystepsize = float(self.txtYstepsize.text())
+            omfdata.zstepsize = float(self.txtZstepsize.text())
+        #convert into sld format
+        omf2sld = sas_gen.OMF2SLD()
+        omf2sld.set_data(omfdata, self.default_shape)
+        sld_data = omf2sld.get_output()
+
+        #only to be done once - load in the position data of the atoms
+        #verification ensures that this is the same across nuclear and magnetic datafiles
+        if self.is_nuc:
+            sld_data.vol_pix = self.nuc_sld_data.vol_pix
+            sld_data.pos_x = self.nuc_sld_data.pos_x
+            sld_data.pos_y = self.nuc_sld_data.pos_y
+            sld_data.pos_z = self.nuc_sld_data.pos_z
+        elif self.is_mag:
+            sld_data.vol_pix = self.mag_sld_data.vol_pix
+            sld_data.pos_x = self.mag_sld_data.pos_x
+            sld_data.pos_y = self.mag_sld_data.pos_y
+            sld_data.pos_z = self.mag_sld_data.pos_z
+
+        #set the sld data from the required model file/GUI textbox
+        if (self.is_nuc):
+            sld_data.set_sldn(self.nuc_sld_data.sld_n)
+        else:
+            sld_data.set_sldn(float(self.txtNucl.text()), non_zero_mag_only=False)
+        if (self.is_mag):
+            sld_data.set_sldms(self.mag_sld_data.sld_mx, self.mag_sld_data.sld_my, self.mag_sld_data.sld_mz)
+        else:
+            sld_data.set_sldms(float(self.txtMx.text()),
+                               float(self.txtMy.text()),
+                               float(self.txtMz.text()))
+        #Provide data giving connections between atoms for 3D drawing
+        #This SHOULD only occur in nuclear data files as it is a feature of
+        #pdb files - however the option for it to be drawn from magnetic files
+        #if present is given in case the sld file format is expanded to include them
+        if self.is_nuc:
+            if self.nuc_sld_data.has_conect:
+                sld_data.has_conect=True
+                sld_data.line_x = self.nuc_sld_data.line_x
+                sld_data.line_y = self.nuc_sld_data.line_y
+                sld_data.line_z = self.nuc_sld_data.line_z
+            elif self.is_mag:
```

Comment:
This should be level with `if self.nuc_sld_data.has_conect:`. It only tries to load in the conect data from the magnetic datafile if it couldn't find it in the nuclear datafile. I believe that with the current set of files there is no way for that data to be found in the magnetic sld data anyway, as only .pdb files store it, however if any feature gets added that allows the line data to be either loaded or transferred to magnetic sld data this catches it so that it can be drawn on 3D plots, if the nuclear data doesn't already contain any. If it becomes a common scenario for both datafiles to contain the conect data a potential enhancement is to combine the two sets of line data fully - so both are displayed on the plot.

Reply:
The extra documentation was useful here. Thanks for the clarification!